games/nethack/Portfile

```diff
@@ -1,110 +1,200 @@
+    if {${subport} eq "nethack"} {
+        reinplace "s|__PREFIX__|${prefix}|" \
+            "${worksrcpath}/sys/unix/Makefile.doc" \
+            "${worksrcpath}/sys/unix/hints/macosx10.10"
+    }
 }
 
-configure.dir       ${worksrcpath}/sys/unix
-configure.cmd       /bin/sh
-configure.pre_args  setup.sh
+configure.dir      ${worksrcpath}/sys/unix
+configure.cmd      ./setup.sh
+configure.pre_args hints/macosx10.10
 
-build.args-append   CC=${configure.cc} \
-                    CXX=${configure.cxx} \
-                    CPP=${configure.cpp} \
-                    CFLAGS="-O2 -I../include"
+use_parallel_build no
+build.args-append  CC=${configure.cc} \
+                   CXX=${configure.cxx} \
+                   CPP=${configure.cpp}
 
 pre-destroot {
     file mkdir "${destroot}${prefix}/share/man/man6"
-    file mkdir "${destroot}${prefix}/share/nethackdir/save"
 }
 
-destroot.target     install manpages
-destroot.keepdirs   "${destroot}${prefix}/share/nethackdir/save/"
+destroot.target    all install manpages
+destroot.keepdirs  "${destroot}${prefix}/var/games/nethack/save/"
 
 post-destroot {
-    reinplace "s|${destroot}||" "${destroot}${prefix}/bin/nethack"
+    if {${subport} eq "nethack"} {
+        reinplace "s|${destroot}||" "${destroot}${prefix}/bin/nethack"
 
-    # Don't overwrite existing preference files
-    foreach f { logfile record } {
-        file rename ${destroot}${prefix}/share/nethackdir/${f} ${destroot}${prefix}/share/nethackdir/${f}.dist
+        # Prevent existing preferences files being overwritten.
+        foreach f { logfile record sysconf } {
+            file rename ${destroot}${prefix}/var/games/nethack/${f} \
+                ${destroot}${prefix}/var/games/nethack/${f}.dist
+        }
     }
 }
 
 post-activate {
-    # Make sure initial preference files exist
-    foreach f { logfile record } {
-        if {![file exists ${prefix}/share/nethackdir/${f}]} {
-            file copy ${prefix}/share/nethackdir/${f}.dist ${prefix}/share/nethackdir/${f}
+    if {${subport} eq "nethack"} {
+        # Make initially-missing preferences files from earlier versions.
+        foreach f { logfile record sysconf } {
+            if {![file exists ${prefix}/var/games/nethack/${f}]} {
+                file copy ${prefix}/var/games/nethack/${f}.dist \
+                    ${prefix}/var/games/nethack/${f}
+            }
         }
     }
 }
 
 variant x11 {
-    patchfiles-append \
-        x11/patch-include__config.h.diff \
-        x11/patch-sys__unix__Makefile.src.diff \
-        x11/patch-sys__unix__Makefile.top.diff
+    patchfiles-append  x11/patch-x11-interface.diff
     depends_lib-append port:xorg-libXaw
 }
 
-variant autopickup_exceptions description { Automatically pick up things onto which you move } {
-    patchfiles-append patch-include__config.h.diff
-}
+subport nethack343 {
+    name                nethack343
+    version             3.4.3
+    revision            4
+    categories          games
```

Comment:
This line can be removed because it comes from the main port.

Reply:
done